dsp-repository/v1/dsp-repository.shacl.ttl

```diff
@@ -26,32 +26,670 @@
+    rdfs:subClassOf schema:Project , prov:Entity .   
+
+### --- Class definition of dsp-repo:Dataset --- 
+# http://ns.dasch.swiss/repository#Dataset
+dsp-repo:Dataset 
+    a rdfs:Class ;
+    rdfs:comment "A dataset"@en ;
+    rdfs:label "dsp-repo:Dataset"@en ;
+    rdfs:subClassOf schema:Dataset .
+
+### --- Class definition of dsp-repo:Person --- 
+# http://ns.dasch.swiss/repository#Person
+dsp-repo:Person 
+    a rdfs:Class ;
+    rdfs:comment "A person"@en ;
+    rdfs:label "dsp-repo:Person"@en ;
+    rdfs:subClassOf schema:Person , prov:Agent . 
+
+### --- Class definition of dsp-repo:Organization --- 
+# http://ns.dasch.swiss/repository#Organization
+dsp-repo:Organization 
+    a rdfs:Class ;
+    rdfs:comment "An organization"@en ;
+    rdfs:label "dsp-repo:Organization"@en ;
+    rdfs:subClassOf schema:Organization , prov:Agent . 
+
+### --- Class definition of dsp-repo:Grant --- 
+# http://ns.dasch.swiss/repository#Grant
+dsp-repo:Grant 
+    a rdfs:Class ;
+    rdfs:comment "A grant"@en ;
+    rdfs:label "dsp-repo:Grant"@en ;
+    rdfs:subClassOf schema:Grant . 
+
+### --- Class definition of dsp-repo:DataManagementPlan --- 
 # http://ns.dasch.swiss/repository#DataManagementPlan
 dsp-repo:DataManagementPlan
-    a owl:Class, sh:NodeShape ;
-    rdfs:label "dsp-repo:DataManagementPlan"@en ;
+    a rdfs:Class ;
     rdfs:comment "A Data Management Plan"@en ;
-    rdfs:subClassOf schema:Thing ;
+    rdfs:label "dsp-repo:DataManagementPlan"@en ;
+    rdfs:subClassOf schema:Thing .
+
+##### SHAPES #####
+### --- Shape of dsp-repo:Project ---
+dsp-repo:ProjectShape
+    a sh:NodeShape ;
+    sh:targetClass dsp-repo:Project ;
+    sh:property [
+        sh:path dsp-repo:hasName ;
+        sh:name "Name" ;
+        sh:description "Name of the project"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+        sh:minCount 1 ;
+    ] ; 
+    sh:property [
+        sh:path dsp-repo:hasDescription ;
+        sh:name "Description" ;
+        sh:description "Description of the project"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+        sh:minCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasKeywords ;
+        sh:name "Keywords" ;
+        sh:description "Keywords and tags"@en ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasDiscipline ;
+        sh:name "Discipline" ;
+        sh:description "Discipline and research fields from UNESCO nomenclature: https://skos.um.es/unesco6/?l=en or from http://www.snf.ch/SiteCollectionDocuments/allg_disziplinenliste.pdf"@en ;
+        sh:or ( [ sh:datatype xsd:string ] [ sh:class schema:URL ] ) ;
+        sh:minCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasStartDate ;
+        sh:name "Start date" ;
+        sh:description "Start date of the project"@en ;
+        sh:datatype xsd:dateTime ;
+        sh:maxCount 1 ;
+        sh:minCount 1 ;
```

Comment:
Input from Rita. I also believe that the start date is a must. It is an indication of how long a project has been running or when it started, for example.

Reply:
I don't think any project should have problems entering a start date.

To ensure this, it should be possible to have inprecise dates (like `1998` or `2012-05`). This would conform with ISO 8601.   
Now, I'm finding conflicting information on that online... would that be valid according to our ontology? Or what do you guys think on that?

In any case: Should it be `xsd:dateTime` or rather `xsd:date`? Projects don't really have a specific start time, or do they?

Reply:
Maybe rather `xsd:date`?

Reply:
changed to `xsd:date`